src/main/scala/RPSService.scala

```diff
@@ -0,0 +1,168 @@
+import scala.io.StdIn.readLine
+import scala.util.Random
+
+class RPSService {
+
+  var rCount1 = 0
+  var pCount1 = 0
+  var sCount1 = 0
+  var rCount2 = 0
+  var pCount2 = 0
+  var sCount2 = 0
+  var computer1Wins = 0
+  var computer2Wins = 0
+  var playerWins = 0
+  var result = ""
+  var computer1Turn = ""
+  var computer2Turn = ""
+  var random = 0.0
+
+  def startPVsAI(): Unit = {
+    println("Hello and welcome to RPS Scala player vs AI.")
+    var playerTurn = validate(readLine("Type end to finish the game, or type r for rock, p for paper, or s for scissors for your turn:"))
+    while (playerTurn != "end") {
+      computer1Turn = takeTurn1()
+      result = determineResultPVsAI(playerTurn, computer1Turn.charAt(0).toString)
+      result match {
+        case "n" => println("It's a draw!")
+        case "c" => println("Computer wins!"); computer1Wins += 1
+        case "p" => println("Player wins!"); playerWins += 1
+        case _ => println("Error! result doesn't match!!")
+      }
+      receive1(playerTurn)
+      println("Current score is computer: " + computer1Wins + " player: " + playerWins)
+      playerTurn = validate(readLine("\nType end to finish the game, or type r for rock, p for paper, or s for scissors for your next turn:"))
+    }
+
+  }
+
+  def validate(choice: String): String = {
+    choice.toLowerCase() match {
+      case "r" => "r"
+      case "rock" => "r"
+      case "p" => "p"
+      case "paper" => "p"
+      case "s" => "s"
+      case "scissors" => "s"
+      case "end" => "end"
+      case _ => validate(readLine("Stop typing like a dickhead. Type r, p or s:"))
+    }
+  }
+
+  def receive1(turn: String): Unit =
+    turn match {
+      case "r" => rCount1 += 1
+      case "p" => pCount1 += 1
+      case "s" => sCount1 += 1
+      case _ => println("Error! Received turn 1 is bad.")
+    }
+
+  def receive2(turn: String): Unit =
+    turn match {
+      case "r" => rCount2 += 1
+      case "p" => pCount2 += 1
+      case "s" => sCount2 += 1
+      case _ => println("Error! Received turn 2 is bad.")
+    }
+
+  def takeTurn1(): String = {
+    val highest = calculateMax1()
```

Comment:
+    val highest = calculateMax1()
what's the porpose of highest here is you're not using it